src/devices/SocketCan/CanFrame.cs

```diff
@@ -0,0 +1,145 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Sockets;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Iot.Device.SocketCan
+{
+    [StructLayout(LayoutKind.Sequential)]
+    public unsafe struct CanFrame
+    {
+        private const int CAN_MAX_DLEN = 8;
+
+        // RawId (can_id) includes EFF, RTR and ERR flags
+        public uint RawId { get; set; }
```

Comment:
>set [](start = 33, length = 3)

Can't you do more validation on the RawId here? or any passed value will be valid?

Reply:
will add some validation (although i.e. setting ERR flag makes frame invalid I will not do validation there as it may be expected - I will only validate that extended frame has ID no larger than 29 bit and standard frame has up to 11 bit id)

Reply:
on the second thought, making this internal